resource-managers/kubernetes/integration-tests/kerberos-yml/data-populator-deployment.yml

```diff
@@ -0,0 +1,49 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
Why Deployment? Should it be a `Job` instead as it runs to completion?

Reply:
This is purely because the job sometimes fails due to networking issues or whatnot and I wanted to counter the flakiness of the test suite by hardening it behind a deployment

Reply:
How does a Deployment helps with that, restarting of failed pod?

Reply:
Yeah, but I am aware that it fits better as a `Job`. (The `hdfs -cp` that is being run is the same file, so if it is re-run it doesn't do any harm.) But if it is a preference to use `Job` I don't see any problems with that as it makes sense